mutt/charset.c

```diff
@@ -314,16 +314,13 @@ int mutt_ch_convert_nonmime_string(char **ps)
   if (ulen == 0)
     return 0;
 
-  const char *c1 = NULL;
-
-  const char *const c_assumed_charset = cs_subset_string(NeoMutt->sub, "assumed_charset");
+  const struct Slist *const c_assumed_charset = cs_subset_slist(NeoMutt->sub, "assumed_charset");
   const char *const c_charset = cs_subset_string(NeoMutt->sub, "charset");
-  for (const char *c = c_assumed_charset; c; c = c1 ? c1 + 1 : 0)
+  const struct ListNode *np;
```

Comment:
```suggestion
  const struct ListNode *np = NULL;
```

Please initialise all pointers.